swarm/api/routes/evolution.py

```
@@ -269,6 +283,19 @@ async def get_evolution_patch_details(
+    try:
+        run_id = validate_path_component(run_id, "run_id")
+        patch_id = validate_path_component(patch_id, "patch_id")
+    except ValueError as e:
+        raise HTTPException(
+            status_code=400,
+            detail={
+                "error": "invalid_path",
+                "message": str(e),
+                "details": {"run_id": run_id, "patch_id": patch_id},
+            },
+        )
```

Ruff (0.15.4)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)